### PR TITLE
[Anomaly] 🐞 NNCF e2e bugfix

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
### Summary

- Fixes e2e issue in nncf

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

## Tests

> _Note:_ This is tested only for Padim but STFPM should work

image

### CODE REVIEW

**Review by samet-akcay — APPROVED** [review]

**Review by samet-akcay — DISMISSED** [review]

File: otx/algorithms/anomaly/tasks/nncf.py [hunk 1]
```diff
@@ -144,9 +144,17 @@ def load_model(self, otx_model: Optional[ModelEntity]) -> AnomalyModule:
                     pl_modules[key] = model_data["model"][key]
             model_data["model"] = nncf_modules
 
+            dataloader = None
```
Comment (jaegukhyun) [inline otx/algorithms/anomaly/tasks/nncf.py]: I think we can add a type hint for this line

File: otx/algorithms/anomaly/tasks/nncf.py [hunk 2]
```diff
@@ -144,9 +144,17 @@ def load_model(self, otx_model: Optional[ModelEntity]) -> AnomalyModule:
...
+            if hasattr(self, "trainer") and hasattr(self.trainer, "datamodule"):
+                if self.trainer.datamodule.train_dataset is not None:
+                    dataloader = self.trainer.datamodule.train_dataloader()
+                elif self.trainer.datamodule.test_dataset is not None:
+                    dataloader = self.trainer.datamodule.test_dataloader()
+
             self.compression_ctrl, model.model = wrap_nncf_model(
                 model.model,
                 self.optimization_config["nncf_config"],
+                dataloader=dataloader,  # type:ignore
```
Comment (jaegukhyun) [inline otx/algorithms/anomaly/tasks/nncf.py:160]: Is it okay dataloader is None? If it is okay then what case could be?

**Review by jaegukhyun — COMMENTED** [review]
I have some minor comments. Could you take a look?

File: otx/algorithms/anomaly/tasks/nncf.py (hunk 2 above)
Comment (ashwinvaidya17) [reply otx/algorithms/anomaly/tasks/nncf.py:160]: If I understand correctly, dataloader will always be None in this case as `wrap_nncf_model` requires either a trained nncf model or a dataloader. In this case we are loading weights from a trained model into nncf. But @AlexanderDokuchaev can clarify this better as he is the owner of `wrap_nncf_model` code

File: otx/algorithms/anomaly/tasks/nncf.py (hunk 1 above)
Comment (ashwinvaidya17) [reply otx/algorithms/anomaly/tasks/nncf.py]: Done!

**Review by jaegukhyun — APPROVED** [review]

**Review by samet-akcay — APPROVED** [review]
thanks @ashwinvaidya17

**Pull request merged**